chain/client/src/client_actor.rs

```diff
@@ -1731,37 +1752,25 @@ impl ClientActor {
                         self.client.runtime_adapter.clone(),
                     )) {
                         StateSyncResult::InProgress => (),
-                        StateSyncResult::RequestBlock => {
-                            if let Some(peer_info) =
-                                self.network_info.highest_height_peers.choose(&mut thread_rng())
-                            {
-                                let id = peer_info.peer_info.id.clone();
-
-                                if let Ok(header) =
-                                    self.client.chain.get_block_header(&state_sync_status.sync_hash)
-                                {
-                                    for hash in
-                                        vec![*header.prev_hash(), *header.hash()].into_iter()
-                                    {
-                                        self.client.request_block(hash, id.clone());
-                                    }
-                                }
-                            }
-                        }
                         StateSyncResult::Completed => {
+                            if !have_block {
+                                info!(target: "sync", "Sync done. Waiting for sync block.");
```

Comment:
this will lead to lots and lots of logspam right? Until the block is applied we will print this out once every 10 millis

Reply:
Changed it to `trace!`. We only enable trace level manually so we should be fine with this one.

Reply:
Hate to press too much on this point... but this doesn't seem like a great solution to me honestly. Basically what we're saying is that with `debug!` it's not great because if you run neard with `RUST_LOG=debug neard run` and hit this particular case, then your logs will be flooded with this message once every 10 ms, making your logs useless and annoying. So now we change it to `trace!` and say that if you run neard with `RUST_LOG=trace neard run` and hit this particular case, then your logs will be flooded with this message once every 10 ms, making your logs useless and annoying.

But this doesn't make a ton of sense to me really, because the only case where we'll even see this log message at all is exactly the case where we hit this bug and the logs get spammed. If we really want to log this, one thing you could do is to check whether each element of `state_sync_status` is `StateSyncDone` before the call to `StateSync::run()`. if not and then that function call returns `Completed`, then we log this